src/odata-to-abstract-sql.ts

```diff
@@ -308,9 +308,10 @@ export class OData2AbstractSQL {
 				return alias;
 			}
 
-			const randStr = randomstring.generate(RANDOM_ALIAS_LENGTH) + '$';
+			const hashStr =
+				stringHash(alias.slice(0, ALIAS_HASH_LENGTH)).toString(36) + '$';
```

Comment:
Wouldn't it be better if we included more characters in the `stringHash` input?
Isn't limiting it to just 6 characters increasing the chance of getting the same `hashStr` for totally different queries?
I realize that
* passing in the whole string might be too expensive with small ROI
* this might not matter given that we still append 56 chars from the actual alias

Reply:
> Isn't limiting it to just 6 characters increasing the chance of getting the same hashStr for totally different queries?

Yes, although the idea was that the hash would be based upon what was being replaced and hence if the same thing is being replaced for two different queries then it should indeed match since the goal is only to shorten and if the unshortened alias would be the same then the shortened alias makes sense to also be the same. What I've realized though is that what we're replacing can definitely be longer than the hash length and so I messed up the selection of the substring to hash - I'll fix that although as you mention it should still be fine due to the 56 chars of actual content.

As a note I knew that passing the full string would also achieve the goal of matching strings matching but I also wanted things like
```
person-has-name
person-has-age
```
to be shortened as eg
```
abcdef$name
abcdef$age
```
if the replaced prefix matches but thinking more on it I think the chances of that happening in reality is too small to be worth the code complexity (it requires the same prefix and the same total length whilst also being >63 chars which is incredibly unlikely)